lldb/source/Plugins/SymbolFile/DWARF/SymbolFileDWARF.cpp

```diff
@@ -4211,6 +4211,9 @@ SymbolFileDWARFDebugMap *SymbolFileDWARF::GetDebugMapSymfile() {
 
 const std::shared_ptr<SymbolFileDWARFDwo> &SymbolFileDWARF::GetDwpSymbolFile() {
   llvm::call_once(m_dwp_symfile_once_flag, [this]() {
+    if (m_objfile_sp->GetArchitecture().GetTriple().isAppleMachO())
```

Comment:
Should this be a check for the ObjectFile subclass instead?

Reply:
@JDevlieghere Maybe you have a better layering suggestion?

Reply:
That was my first inclination. But then I was reading the discussion at https://github.com/llvm/llvm-project/pull/139170#issuecomment-2871870612 and was going to see where that goes

Reply:
Yeah, this sounds like very much a dual problem to that, so it would be nice if they used the same approach. The thing that makes this fuzzier is that in this case you can't really say that SymbolFileDWARF *requires* ObjectFileMachO (or any other) because one can imagine a setup where someone knows they will only ever need to debug DWARF+ELF and so they want to strip out all of the other object file plugins. OTOH, I can also imagine us saying we don't want to support that. :P

Reply:
Not sure what the resolution was re. the plugin dependencies discussion. Are we find with the current approach?

> you can't really say that SymbolFileDWARF requires ObjectFileMachO (or any other) because one can imagine a setup where someone knows they will only ever need to debug DWARF+ELF and so they want to strip out all of the other object file plugins.

Could question. If we don't want to restrict a use-case such as the one you describe, I'm happy to add a `SupportsDWP` API to ObjectFile

Reply:
I think this is a good solution until we've documented the dependency policy from #139170